video-games/video-games-test.js

```diff
@@ -0,0 +1,105 @@
+   });
+
+   it.skip('should be able to create a new level with a name and many players', function () {
+      var player1 = createPlayer('Rock Star Rebel', 21, ['power chord', 'stage dive', 'crowd surf']);
+      var player2 = createPlayer('Pro Skater Prodigy', 16, ['360 flip', 'board slide', 'gap transfer']);
+      var level = createLevel('Candy Cane Canyon', [player1, player2]);
+
+      assert.equal(level.name, 'Candy Cane Canyon');
+      assert.deepEqual(level.players, [player1, player2]);
+   });
+
+   it.skip('should initiate a level with a player and 3 lives and 0 coins', function () {
+      var player1 = createPlayer('Quantum Jumper', 30, ['dimension jump', 'portal creation', 'alternate reality explosion']);
+      var level = createLevel('Carnival of Terror', [player1]);
+
+      assert.equal(level.name, 'Carnival of Terror');
+      assert.deepEqual(level.players, [player1]);
+      assert.equal(level.coins, 0);
+      assert.equal(level.lives, 3);
+   });
+
+   it.skip('should be able to collect coins', function () {
+      var player1 = createPlayer('The Timeless Troubadour', 500, ['piano sonata', 'violin concerto', 'fllamenco guitar']);
+      var level = createLevel('Infernal Inferno of Flames', [player1]);
+
+      assert.equal(level.name, 'Infernal Inferno of Flames');
+      assert.deepEqual(level.players, [player1]);
+      assert.equal(level.coins, 0);
+      assert.equal(level.lives, 3);
+
+      var updatedLevel = findCoins(level, 5);
+      assert.equal(updatedLevel.coins, 5);
+   });
+
+   it.skip('should be able to defeat the player and cost a life', function () {
+      var player1 = createPlayer('Tony Thrasher', 34, ['thrash throw', 'solving mysteries', 'thrash dance']);
+      var level = createLevel('Alien Outpost', [player1]);
+
+      assert.equal(level.name, 'Alien Outpost');
+      assert.deepEqual(level.players, [player1]);
+      assert.equal(level.coins, 0);
+      assert.equal(level.lives, 3);
+
+      var updatedLevelPlayerFell = defeatPlayer(level);
+      assert.equal(updatedLevelPlayerFell.lives, 2);
+   }); 
+
+   it.skip('should print `GAME OVER` when a player is defeated and has no lives left', function () {
+      var player1 = createPlayer('Ant Cat', 3, ['tiny roar', 'six-legged claws', 'dig tunnel']);
+      var level = createLevel('Alien Outpost', [player1]);
+
+      assert.equal(level.name, 'Alien Outpost');
+      assert.deepEqual(level.players, [player1]);
+      assert.equal(level.coins, 0);
+      assert.equal(level.lives, 3);
+
+      var updatedLevelPlayerDefeated = defeatPlayer(level);
+      assert.equal(updatedLevelPlayerDefeated.lives, 2);
+      var updatedLevelPlayerAnnihilated  = defeatPlayer(updatedLevelPlayerDefeated);
+      assert.equal(updatedLevelPlayerAnnihilated.lives, 1);
+      var updatedLevelPlayerDestroyed = defeatPlayer(updatedLevelPlayerAnnihilated);
+      assert.equal(updatedLevelPlayerDestroyed, 'GAME OVER');
+   }); 
```

Comment:
lol @ these variable names

Reply:
Trying to enforce some FP! I agree these names are a bit contrived